k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/SelectOrExamineCommand.java

```diff
@@ -0,0 +1,63 @@
+package com.fsck.k9.mail.store.imap;
+
+
+import static com.fsck.k9.mail.Folder.OPEN_MODE_RW;
+
+
+class SelectOrExamineCommand {
+
+    private static final long INVALID_UIDVALIDITY = -1L;
+    private static final long INVALID_HIGHESTMODSEQ = -1L;
+    private static final long INVALID_SMALLEST_UID = -1L;
+
+    private int mode;
+    private String escapedFolderName;
+    private boolean useCondstore;
+    private long cachedUidValidity;
+    private long cachedHighestModSeq;
+    private long smallestUid;
+
+    static SelectOrExamineCommand createNormal(int mode, String escapedFolderName) {
+        return new SelectOrExamineCommand(mode, escapedFolderName, false, INVALID_UIDVALIDITY, INVALID_HIGHESTMODSEQ,
+                INVALID_SMALLEST_UID);
+    }
+
+    static SelectOrExamineCommand createWithCondstoreParameter(int mode, String escapedFolderName) {
+        return new SelectOrExamineCommand(mode, escapedFolderName, true, INVALID_UIDVALIDITY, INVALID_HIGHESTMODSEQ,
+                INVALID_SMALLEST_UID);
+    }
+
+    static SelectOrExamineCommand createWithQresyncParameter(int mode, String escapedFolderName,
+            long cachedUidValidity, long cachedHighestModSeq, long smallestUid) {
```

Comment:
It's not really too important since the parameter is not used if QRESYNC is used, but I see your point. I'll change this.